datafusion/core/src/physical_plan/mod.rs

```diff
@@ -42,6 +42,80 @@ use std::sync::Arc;
 use std::task::{Context, Poll};
 use std::{any::Any, pin::Pin};
 
+/// Enum representation of a physical plan
+pub enum PhysicalPlan {
```

Comment:
😍 

The only thing that is a little strange to me is that the compiler would allow `PhysicalPlan::Filter` to have something that wasn't actually a filter (aka it could be `HashJoinExec`)